lib/utils/checkAgainstRule.js

```diff
@@ -3,39 +3,45 @@
 const normalizeRuleSettings = require('../normalizeRuleSettings');
 const Result = require('postcss/lib/result');
 const rules = require('../rules');
+const { isPlainObject } = require('./validateTypes');
 
 /**
  * Useful for third-party code (e.g. plugins) to run a PostCSS Root
  * against a specific rule and do something with the warnings
  * @template T
  * @template {Object} O
  * @param {{
-		ruleName: string,
-		ruleSettings: import('stylelint').ConfigRuleSettings<T, O>,
-		root: import('postcss').Root,
-	}} options
+ *   ruleName: string,
+ *   ruleSettings: import('stylelint').ConfigRuleSettings<T, O>,
+ *   root: import('postcss').Root,
+ *   result?: import('stylelint').PostcssResult,
+ * }} options
  * @param {(warning: import('postcss').Warning) => void} callback
  * @returns {void}
  */
 function checkAgainstRule(options, callback) {
-	if (!options)
-		throw new Error(
-			"checkAgainstRule requires an options object with 'ruleName', 'ruleSettings', and 'root' properties",
-		);
+	if (!isPlainObject(options)) throw new Error('checkAgainstRule requires an options object');
 
 	if (!callback) throw new Error('checkAgainstRule requires a callback');
 
-	if (!options.ruleName) throw new Error("checkAgainstRule requires a 'ruleName' option");
+	const { ruleName, ruleSettings, root, result } = options;
 
-	const rule = rules[options.ruleName];
+	if (!ruleName) throw new Error("checkAgainstRule requires a 'ruleName' option");
 
-	if (!rule) throw new Error(`Rule '${options.ruleName}' does not exist`);
+	const rule =
+		rules[ruleName] ||
+		(result &&
+			result.stylelint.config &&
+			result.stylelint.config.pluginFunctions &&
+			result.stylelint.config.pluginFunctions[ruleName]);
```

Comment:
[suggest] I think it's time to reduce the following duplication:

- https://github.com/stylelint/stylelint/blob/35c0cfa244f725a3548e8ba884edf70c2bc67732/lib/normalizeAllRuleSettings.js#L20
- https://github.com/stylelint/stylelint/blob/35c0cfa244f725a3548e8ba884edf70c2bc67732/lib/lintPostcssResult.js#L65-L66

So what about extracting the following utility function (`utils/getStylelintRule.js`)?

```js
/**
 * @param {string} ruleName
 * @param {import('stylelint').Config | undefined} [config]
 * @returns {import('stylelint').Rule | undefined}
 */
module.exports = function getStylelintRule(ruleName, config) {
	return rules[ruleName] || (config && config.pluginFunctions && config.pluginFunctions[ruleName]);
};
```

Reply:
Big fan of this clean up: https://github.com/stylelint/stylelint/pull/6460/commits/339005b8d3072a8b6708387c76ce5d9aec4b7e95